### PR TITLE
[LegalizeVectorOps] Enable ExpandFABS/COPYSIGN to use integer ops for fixed vectors in some cases.

### PR DESCRIPTION
**Pull request by topperc**
Copy the same FSUB check from ExpandFNEG to avoid breaking AArch64 and ARM.

### CODE REVIEW

File: llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp [hunk 1]
```diff
@@ -1840,7 +1847,12 @@ SDValue VectorLegalizer::ExpandFCOPYSIGN(SDNode *Node) {
   // FIXME: We shouldn't restrict this to scalable vectors.
```
Comment (justinfargnoli) [inline llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp:1847]: It seems like this comment should also be removed.

File: llvm/test/CodeGen/NVPTX/f16x2-instructions.ll [hunk 2]
```diff
@@ -1184,14 +1184,15 @@ define <2 x half> @test_fma(<2 x half> %a, <2 x half> %b, <2 x half> %c) #0 {
...
-; CHECK-DAG:  cvt.f32.f16     [[AF0:%f[0-9]+]], [[A0]];
-; CHECK-DAG:  cvt.f32.f16     [[AF1:%f[0-9]+]], [[A1]];
-; CHECK-DAG:  abs.f32         [[RF0:%f[0-9]+]], [[AF0]];
-; CHECK-DAG:  abs.f32         [[RF1:%f[0-9]+]], [[AF1]];
-; CHECK-DAG:  cvt.rn.f16.f32  [[R0:%rs[0-9]+]], [[RF0]];
-; CHECK-DAG:  cvt.rn.f16.f32  [[R1:%rs[0-9]+]], [[RF1]];
-; CHECK:      mov.b32         [[R:%r[0-9]+]], {[[R0]], [[R1]]}
+; CHECK-NOF16:      mov.b32         {[[A0:%rs[0-9]+]], [[A1:%rs[0-9]+]]}, [[A]]
+; CHECK-NOF16-DAG:  cvt.f32.f16     [[AF0:%f[0-9]+]], [[A0]];
+; CHECK-NOF16-DAG:  cvt.f32.f16     [[AF1:%f[0-9]+]], [[A1]];
```
Comment (arsenm) [inline llvm/test/CodeGen/NVPTX/f16x2-instructions.ll:1189]: This still has the unwanted conversions, I assume from type legalization

File: llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp [hunk 3]
```diff
@@ -1821,8 +1824,12 @@ SDValue VectorLegalizer::ExpandFABS(SDNode *Node) {
   EVT VT = Node->getValueType(0);
   EVT IntVT = VT.changeVectorElementTypeToInteger();
 
-  // FIXME: We shouldn't restrict this to scalable vectors.
-  if (!TLI.isOperationLegalOrCustom(ISD::AND, IntVT) || !VT.isScalableVector())
+  if (!TLI.isOperationLegalOrCustom(ISD::AND, IntVT))
+    return SDValue();
+
+  // FIXME: The FSUB check is here to force unrolling v1f64 vectors on AArch64.
```
Comment (arsenm) [inline llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp:1830]: can you really not just delete this now? There are no more fsubs emitted

Comment (topperc) [reply llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp:1830]: It's a proxy for FP arithmetic not being supported. AArch64 has a v1f64 as a legal type but Expands all operations. Using v1i64 for FNEG and scalarizing other arithmetic causes obvious regressions in tests.

**Comment by topperc** [comment]
Ping

**Review by arsenm — APPROVED** [review]

**Pull request merged**